pkg/solana/logpoller/log_poller.go

```diff
@@ -0,0 +1,150 @@
+package logpoller
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+var (
+	ErrFilterNameConflict = errors.New("filter with such name already exists")
+)
+
+type ORM interface {
+	InsertFilter(ctx context.Context, filter Filter) (id int64, err error)
+	SelectFilters(ctx context.Context) ([]Filter, error)
+	DeleteFilters(ctx context.Context, filters map[int64]Filter) error
+	MarkFilterDeleted(ctx context.Context, id int64) (err error)
+}
+
+type LogPoller struct {
```

Comment:
Isn't naming the struct after the component rather than just `Service` the convention we have in the whole Chainlink codebase, not just `LogPoller`? We do the same thing for `ChainReader`, `ChainWriter`, `Multinode`, every component I can think of. If there are any that don't follow this pattern I'm unaware of them.

Also, the long-term plan is to have the generalized LogPoller framework specified in `chainlink-common` in the form of a `LogPoller` interface that has to be implemented. Wouldn't it be a bit strange to have it named `Service` when it's implementing `LogPoller`?

Reply:
What if we named the package `lp` so that it wasn't as redundant?

Reply:
The only thing I notice that's a bit different is that it's exported... is there a reason why it has to be exported? Usually the struct itself is `logPoller` and only the interface is capitalized. Is this is exported to make testing easier?

Reply:
As this pkg does not provide multiple logpoller implementation there is no reason to define interface and we should export struct. In our codebase, we tend to declare interfaces in pkgs with implementation, while go's idiomatic way is to declare the interface where it's used.